tests/common/plugins/test_completeness/__init__.py

```diff
@@ -4,10 +4,37 @@
 import warnings
 
 class CompletenessLevel(enum.IntEnum):
-    debug = 0 # Minimum execution
-    basic = 1
-    confident = 2
-    thorough = 3 # Maximum execution
+    diagnose = 0
+    debug = 1 # Minimum execution
+    basic = 2
+    confident = 3
+    thorough = 4 # Maximum execution 
+
+    @classmethod
+    def get_normalized_level(cls, request):
+        """Takes a request instance and returns normalized completeness level in string format.
+        For example, if a testcase supports "CompletenessLevel.basic, CompletenessLevel.thorough", and specified level
+        during test execution is "confident", this method will make use of normalization logic during
+        pytest_runtest_setup and returns the normalized level as "basic" (str type of CompletenessLevel.basic)
+        """
+        all_supported_levels = [mark.args for mark in request.node.iter_markers(name="supported_completeness_level")]
+        logging.info("All supported completeness levels of the test: {}".format(str(all_supported_levels)))
+        normalized_level = all_supported_levels[0][0]
+        normalized_level_name = CompletenessLevel.get_level_name(normalized_level)
+        logging.info("Normalized completeness level set to: {}".format(normalized_level_name))
+        return normalized_level_name
+
+    @classmethod
+    def get_level_name(cls, level):
+        """Converts a type CompletenessLevel to type str.
+        For example, if input is CompletenessLevel.basic, this method will return "basic"
+        Arguments:
+        level - An enum value of type CompletenessLevel
```

Comment:
Just some minor formatting changes:
```suggestion
        """Converts a type CompletenessLevel to type str.
        
        For example, if input is CompletenessLevel.basic, this method will return "basic."
        
        Arguments:
            level - An enum value of type CompletenessLevel
```

Reply:
Incorporated the suggestion. Thanks!